tests/unit/test_token_response.py

```diff
@@ -0,0 +1,236 @@
+import requests
+import json
+import six
+import time
+import unittest
+try:
+    import jose
+    JOSE_FLAG = True
+except ImportError:
+    JOSE_FLAG = False
+try:
+    import mock
+except ImportError:
+    from unittest import mock
+
+from tests.framework import (CapturedIOTestCase, SDKTESTER1A_NATIVE1_ID_TOKEN,
+                             SDKTESTER1A_ID_ACCESS_TOKEN, get_client_data)
+from globus_sdk.auth.token_response import (
+    OAuthTokenResponse, OAuthDependentTokenResponse, _convert_token_info_dict)
+from globus_sdk.exc import GlobusOptionalDependencyError
+
+
+class OAuthTokenResponseTests(CapturedIOTestCase):
+
+    def setUp(self):
+        """
+        Creates a OAuthTokenResponse object with known data for testing
+        Sets up mock AuthClient for decoding id_tokens
+        """
+        super(OAuthTokenResponseTests, self).setUp()
+        # known token data for testing expected values
+        self.top_token = {  # valid id_token and access_token
+            "resource_server": "server1", "expires_in": 10, "scope": "scope1",
+            "refresh_token": "RT1", "other_tokens": [], "token_type": "1",
+            "id_token": SDKTESTER1A_NATIVE1_ID_TOKEN,
+            "access_token": SDKTESTER1A_ID_ACCESS_TOKEN}
+        self.other_token1 = {  # invalid id_token with valid access_token
+            "resource_server": "server2", "expires_in": 20, "scope": "scope2",
+            "refresh_token": "RT2", "other_tokens": [], "token_type": "2",
+            "id_token": "invalid_id_token",
+            "access_token": SDKTESTER1A_ID_ACCESS_TOKEN}
+        self.other_token2 = {  # valid id_token with invalid access_token
+            "resource_server": "server3", "expires_in": 30, "scope": "scope3",
+            "refresh_token": "RT3", "other_tokens": [], "token_type": "3",
+            "id_token": SDKTESTER1A_NATIVE1_ID_TOKEN,
+            "access_token": "invalid_access_token"}
+        self.top_token["other_tokens"] = [self.other_token1, self.other_token2]
+
+        # create the response
+        http_response = requests.Response()
+        http_response._content = six.b(json.dumps(self.top_token))
+        http_response.headers["Content-Type"] = "application/json"
+        self.response = OAuthTokenResponse(http_response)
```

Comment:
This is not a "right now" issue, but maybe it would be nice to have a helper in `tests.framework` which converts a dict to a response, given a response class. Just thought of it seeing this a couple of times in this PR.